documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc

```diff
@@ -482,7 +482,7 @@ completes.
 [id='rest_v2_container_configuration_listen']
 = Listening to container configuration events
 Receive events about configuration changes using https://html.spec.whatwg.org/multipage/server-sent-events.html[Server-Sent Events].
-The `event` value will be one of `cache-created`, `cache-removed`.
+The `event` value will be one of `create-cache`, `remove-cache` or `update-cache`.
```

Comment:
it looks like we have events for templates too:
`create-template` and `remove-template`

Reply:
We have events for `remove-template`. I don't think we have `create-template` because the `@Listener` implementation is as follows:

```java
      @ConfigurationChanged
      public CompletionStage<Void> onConfigurationEvent(ConfigurationChangedEvent event) {
         String eventType = event.getConfigurationEventType().toString().toLowerCase() + "-" + event.getConfigurationEntityType();
         final ServerSentEvent sse;
         if (event.getConfigurationEventType() == ConfigurationChangedEvent.EventType.REMOVE) {
            sse = new ServerSentEvent(eventType, event.getConfigurationEntityName());
         } else {
            switch (event.getConfigurationEntityType()) {
               case "cache":
                  try {
                     sse = new ServerSentEvent(eventType, cacheConfig(cacheManager, event.getConfigurationEntityName(), mediaType));
                  } catch (Exception e) {
                     Log.REST.errorf(e, "Could not serialize cache configuration");
                     return CompletableFutures.completedNull();
                  }
                  break;
               default:
                  // Unhandled entity type, ignore
                  return CompletableFutures.completedNull();
            }
         }
         return eventStream.sendEvent(sse);
      }
   }
```
@tristantarrant I guess that's a bug we should address here as well?

Reply:
YEs please

Reply:
Updated. AFAIK there's no way to delete a template via REST, so the config listener test doesn't cover these events only template creation.